pkg/wallet/wallet.go

```diff
@@ -138,51 +140,25 @@ func (w *BaseWallet) CreateDID(method string, opts ...DocOpts) (*did.Doc, error)
 	}
 
 	// Generate key pair
-	pub, err := w.CreateEncryptionKey()
+	base58PubKey, err := w.CreateEncryptionKey()
 	if err != nil {
 		return nil, fmt.Errorf("failed to create DID: %w", err)
 	}
 
-	// DID identifier
-	id := fmt.Sprintf(didFormat, method, pub[:16])
-
-	// Supporting only one public key now
-	pubKey := did.PublicKey{
-		ID: fmt.Sprintf(didPKID, id, 1),
-		// TODO hardcoding public key type for now
-		// Should be dynamic for multi-key support
-		Type:       "Ed25519VerificationKey2018",
-		Controller: id,
-		Value:      []byte(pub),
-	}
+	var didDoc *did.Doc
 
-	// Service model to be included only if service type is provided through opts
-	var service []did.Service
-	if docOpts.serviceType != "" {
-		// Service endpoints
-		service = []did.Service{
-			{
-				ID:              fmt.Sprintf(didServiceID, id, 1),
-				Type:            docOpts.serviceType,
-				ServiceEndpoint: w.inboundTransportEndpoint,
-			},
+	switch method {
+	case peerDIDMethod:
+		didDoc, err = w.buildPeerDIDDoc(base58PubKey, docOpts)
+		if err != nil {
+			return nil, fmt.Errorf("create peer DID : %w", err)
 		}
-	}
-
-	// Created time
-	createdTime := time.Now()
-
-	didDoc := &did.Doc{
-		Context:   []string{did.Context},
-		ID:        id,
-		PublicKey: []did.PublicKey{pubKey},
-		Service:   service,
-		Created:   &createdTime,
-		Updated:   &createdTime,
+	default:
+		didDoc = w.buildDefaultDIDDoc(method, base58PubKey, docOpts)
```

Comment:
what's a default DID doc?

Reply:
Thats the existing code and I just added a switch for PeerDID. @llorllale can you confirm if we need to support this ?

Reply:
@troyronda removed the default case.